### DISCUSSION TITLE
Window is not defined in vercel deploy

### DISCUSSION THREAD

**Discussion by mortske** [vercel/next.js#40444] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Home
    Binaries:
      Node: 16.15.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.3.0
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0

### How are you deploying your application? (if relevant)

Vercel

### Describe the Bug

I am using an npm package called react-leaflet, which expects window to exist. I have added next/dynamic with ssr: false, which works locally, but when deploying this to vercel i still get "ReferenceError: window is not defined"
![image](https://user-images.githubusercontent.com/10628742/189535217-dce37b5a-fbc4-4421-a3d8-8b213ea2f90f.png)

Code usage:
/index.tsx
`import Map from './Components/Map/DynamicMap'
export default function Home() {
  return (
    <div className="container">
      <Head>
        <title>Create Next App</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <main>
        {
          <Map />
        }
      </main>
    </div>
  )
}`

Components/Map/DynamicMap.tsx
`import dynamic from 'next/dynamic';

const DynamicMap = dynamic(() => import('./Map'), {
  ssr: false
});

export default DynamicMap;`

Components/Map/Map.tsx
`import { MapContainer, Marker, Popup, TileLayer } from 'react-leaflet'
import 'leaflet/dist/leaflet.css'

const Map = () => {
  return (
    <MapContainer center={[77.9, -86.1]} zoom={4} maxZoom={6} scrollWheelZoom={true} style={{height: "100vh", width: "100vw"}}>
      <TileLayer
        url="/tiles/{z}/{x}/{y}.png"
      />
      <Marker position={[77.9, -86.1]}>
        <Popup>
          A pretty CSS3 popup. <br /> Easily customizable.
        </Popup>
      </Marker>
    </MapContainer>
  )
}
export default Map`

### Expected Behavior

Expecting the build to work, the same way it works locally

### Link to reproduction

https://codesandbox.io/embed/m7qnb

### To Reproduce

Reproduction link is just a similar setup to what causes my build to fail.

Upload above code to a vercel project
Wait for deploy to fail

**Comment by Uny1me** [comment]
It should replicate on local as well, did you try 
if (typeof window !== 'undefined') and put the follow up code in the block?

**Comment by mortske** [comment]
Yes, i tried that, same result. Worked in local, but failed in the build

**Reply by icyJoseph** [reply]
Worked in local? `next dev` is one thing, but you should also run `next build` and `next start` locally.

Some packages actually ship different code for production and development, not claiming this is the case with `react-leaftlet`, but as a developer you ought to check all transpiling steps (both dev and prod), before deploying.

### Edit

It builds locally for me too, so now the plot thickens...

**Comment by icyJoseph** [comment]
I've tried to break this in a number of ways, but I can't, it always builds. 

I found this issue: https://github.com/Leaflet/Leaflet/issues/8327, but other than that I cannot break it. At least not the https://codesandbox.io/embed/m7qnb example you posted. 

I even updated some dependencies to try and break it and nothing... could you please try to create a [stackblitz](https://stackblitz.com/) reproduction? **_Or even better, a link to a repository._**

**Comment by mortske** [comment]
I seem to have found the issue, i realised my Components folder (where i stored my Map component) was located in the pages folder. I tried moving it out to the root and the build went through when i ran it locally

**Reply by icyJoseph** [reply]
Ah yeah, because it was trying to build the components route. Good job!
